src/SqlStreamStore.Postgres/PostgresStreamStore.ReadAll.cs

```diff
@@ -18,68 +23,363 @@ partial class PostgresStreamStore
+            }
+
+
+            for(int i = 0; i < messages.Count - 1; i++)
+            {
+                var expectedNextPosition = messages[i].Position + 1;
+                var actualPosition = messages[i + 1].Position;
+                Logger.TraceFormat(
+                    "{correlation} Gap checking. Expected position: {expectedNextPosition} | Actual position: {actualPosition}",
+                    correlation,
+                    expectedNextPosition,
+                    actualPosition);
+
+                if(expectedNextPosition != actualPosition)
+                {
+                    Logger.TraceFormat("{correlation} Gap detected", correlation);
+
+                    await PollTransactions(correlation, transactionsInProgress, cancellationToken);
+
+                    return await ReadTrustedMessages(fromPositionInclusive, messages[messages.Count - 1].Position, maxCount, prefetch, correlation, cancellationToken);
+                }
+            }
+
+            if(IsEnd(readAllResult.Count, maxCount))
+                return (readAllResult, true);
+
+            readAllResult.RemoveAt(maxCount);
+            return (readAllResult, false);
+        }
+
+        private async Task PollTransactions(
+            Guid correlation,
+            TxIdList transactionsInProgress,
+            CancellationToken cancellationToken)
+        {
+            Logger.TraceFormat(
+                "{correlation} Transactions {transactions} are in progress, so gaps might be filled, start comparing",
+                correlation,
+                transactionsInProgress);
+
+            bool stillInProgress;
+            int count = 1;
+            int delayTime = 10;
+            long totalTime = 0;
+
+            var sw = Stopwatch.StartNew();
+            do
+            {
+                stillInProgress = await ReadAnyTransactionsInProgress(transactionsInProgress, cancellationToken);
+                Logger.TraceFormat(
+                    "{correlation} Transactions still pending. Query 'ReadAnyTransactionsInProgress' took: {timeTaken}ms",
+                    correlation,
+                    sw.ElapsedMilliseconds);
+
+                if(_settings.GapHandlingSettings.AllowSkip
+                   && totalTime > _settings.GapHandlingSettings.PossibleAllowedSkipTime)
+                {
+                    Logger.ErrorFormat(
+                        "{correlation} Possible SKIPPED EVENT as we will stop waiting for in progress transactions! One of these transactions: {transactions} is in progress for longer then {totalTime}ms",
+                        correlation,
+                        transactionsInProgress,
+                        _settings.GapHandlingSettings.PossibleAllowedSkipTime);
+                    return;
+                }
+
+                if(totalTime > _settings.GapHandlingSettings.PossibleDeadlockTime)
```

Comment:
I'm not sure if I'm following the name.
So it will start logging after `PossibleDeadlockTime` but do nothing right?

Wouldn't 'MinimumWarnTime' be better suited or something else?
Since deadlocking is something that never completes while transactions will complete eventually.